picky-asn1-der/src/application_tag.rs

```diff
@@ -0,0 +1,129 @@
+use crate::misc::Length;
+use crate::Asn1RawDer;
+use picky_asn1::tag::{Tag, TagPeeker};
+use serde::de::{Error, SeqAccess};
+use serde::{de, ser};
+use std::fmt;
+use std::fmt::Debug;
+
+#[derive(Debug, PartialEq)]
+pub struct ApplicationTag<V: Debug + PartialEq, const T: u8>(V);
```

Comment:
Also, do you remember https://github.com/Devolutions/picky-rs/pull/92 and https://github.com/Devolutions/picky-rs/issues/91?
If I understand correctly you did a generic implementation of `ExplicitContextTag` (you need to rename from `ApplicationTag`)